test/kubernetes_unittest.cc

```diff
@@ -72,27 +77,296 @@ class KubernetesTest : public ::testing::Test {
       throw(json::Exception) {
     return reader->UpdateServiceToPodsCache(endpoints, is_deleted);
   }
+
+  void SetUp() override {
+    config = CreateConfig();
```

Comment:
How is this going to work when we have to do other computations before creating the config? For example, the tests using the fake HTTP server first spin up the server, then add its address to the config string.

Reply:
I would imagine you'd override `SetUp` to do the necessary pre-work, and override `CreateConfig` to use the values created by the pre-work in creating the config object, e.g.:
```
void SetUp() override {
  fake_server.reset(new FakeHttpServer(...));
  KubernetesTest::SetUp();
}

std::unique_ptr<Configuration> CreateConfig() override {
  return std::unique_ptr<Configuration>(
    new Configuration(std::istringstream(
      std::string("SomeServiceUrl: \"") + fake_server.GetUrl() + "\"\n"
    )));
}
```